app/controllers/developer_controller.rb

```diff
@@ -13,7 +13,28 @@ def update_sites
 
   def production_log
     @log = if params[:grep].present?
-             `tail -n 10000 log/production.log | grep -E '#{params[:grep]}' -C #{params[:context]} --color=never`
+             grep_str = if params[:grep].instance_of?(String)
+                          params[:grep]
+                        elsif params[:grep].respond_to?(to_s)
+                          params[:grep].to_s
+                        else
+                          flash[:warning] = 'Coercion failure on params[:grep]'
+                          redirect_to(root_path) && return
+                        end
```

Comment:
You also don't need any of this error handling at all, since `params[:grep]` is always going to be a string, hash, or array, all of which respond to `to_s`, and in fact will automatically be coerced to a string in the string interpolation on line 37.